src/reducers/initialState.js

```diff
@@ -0,0 +1,10 @@
+import Immutable from 'immutable';
+
+export default {
+  versions: [],
+  histories: Immutable.Map(),
```

Comment:
Do you mean in contrast to versions? In applicationVersionsReducer I return a new array everytime. So state is not mutated here.